### PR TITLE
Switch to Using the -hadoop3 HBase Artifacts Where Available

### PR DESCRIPTION
**Pull request by stoty**
bump HBase 2.5 version to 2.5.2-hadoop3

### CODE REVIEW

File: BUILDING.md [hunk 1]
```diff
@@ -32,8 +32,13 @@ HBase 2 and Hadoop 3
 Phoenix 5.x requires Hadoop 3. While HBase 2.x is compatible with Hadoop 3, the public Maven HBase
 artifacts are built with Hadoop 2, and are not.
 
-For this reason, when building Phoenix, you need to rebuild
-HBase with Hadoop 3, and install it to the local maven repo of the build host.
+With the 2.5.2 release HBase has started releasing Maven artifacts built with Hadoop3.
+Where available, use the Maven artifacts with the `-hadoop3` postfix in the version,
+i.e. `2.5.2-hadoop3`, and ignore the rest of this section.
+For Hbase 2.5, Phoenix already uses the -hadoop3 version by default.
```
Comment (gjacoby126) [inline BUILDING.md:38]: tiny nit: the B in HBase is capitalized.

**Review by gjacoby126 — APPROVED** [review]
+1 assuming tests pass. Is there follow-up work to do in the release management / voting scripts?

**Comment by stoty** [comment]
>  Is there follow-up work to do in the release management / voting scripts?

Nothing I can think of.
It's as simple as changing the version the HBase version number, and I've added the bypass to the Hbase rebuild helper script.
The HBase version number is not exposed in the Phoenix artifact names, we only use the profile for that.

File: BUILDING.md (hunk 1 above)
Comment (stoty) [reply BUILDING.md:38]: Thanks, will fix before commit.

**Review by virajjasani — APPROVED** [review]

**Comment by gjacoby126** [comment]
This has been merged so I'm closing the PR.

**Pull request closed without merging**